pkg/api/store/cluster/cluster_store.go

```diff
@@ -941,3 +948,47 @@ func replaceWithOldSecretIfNotExists(oldData, newData map[string]interface{}, cl
 		values.PutValue(newData, oldSecret, keys...)
 	}
 }
+
+func (r *Store) validateUnavailableNodes(data, existingData map[string]interface{}, id string) error {
+	cluster, err := r.ClusterLister.Get("", id)
+	if err != nil {
+		return fmt.Errorf("error getting cluster, try again %v", err)
+	}
+	// no need to validate if cluster's already provisioning or upgrading
+	if v3.ClusterConditionProvisioned.IsUnknown(cluster) ||
+		v3.ClusterConditionUpgraded.IsUnknown(cluster) ||
+		v3.ClusterConditionUpdated.IsUnknown(cluster) {
+		return nil
+	}
+	getRkeConfig := func(data map[string]interface{}) (*v3.RancherKubernetesEngineConfig, error) {
```

Comment:
is there a reason to write this as a function literal? I feel the significantly increase the complexity and reduce the readability of the code.